### PR TITLE
Fix bug that the uninstaller may get stuck there

### PR DESCRIPTION
**Pull request by shuo-wu**
longhorn/longhorn#1820

### CODE REVIEW

File: controller/uninstall_controller.go [hunk 1]
```diff
@@ -70,8 +70,8 @@ func NewUninstallController(
 	c := &UninstallController{
 		baseController: newBaseControllerWithQueue("longhorn-uninstall", logger,
 			workqueue.NewNamedRateLimitingQueue(workqueue.NewMaxOfRateLimiter(
-				workqueue.NewItemExponentialFailureRateLimiter(500*time.Millisecond, 5*time.Second),
-				&workqueue.BucketRateLimiter{Limiter: rate.NewLimiter(rate.Limit(10), 100)},
+				workqueue.NewItemExponentialFailureRateLimiter(100*time.Millisecond, 2*time.Second),
```
Comment (yasker) [inline controller/uninstall_controller.go:73]: We can just use `EnhancedDefaultControllerRateLimiter` unless it doesn't fit for this scenario.

**Review by yasker — CHANGES_REQUESTED** [review]

File: controller/uninstall_controller.go (hunk 1 above)
Comment (shuo-wu) [reply controller/uninstall_controller.go:73]: The parameter (max delay is 1000s) of `NewItemExponentialFailureRateLimiter` in `EnhancedDefaultControllerRateLimiter` doesn't fit the uninstaller. You can see the issue post for details.

Comment (yasker) [reply controller/uninstall_controller.go:73]: Should we have a max delay of 1000s for other controllers? It was copied from `DefaultControllerRateLimiter`.

Comment (shuo-wu) [reply controller/uninstall_controller.go:73]: In other controllers, the keys are typically different. Hence the `EnhancedDefaultControllerRateLimiter ` should be fine.
But in the uninstall controller, the key is always `uninstall`. Keeping enqueuing the same key will be considered as handling failure then trigger the exponential backoff. Then the uninstaller may get blocked.

**Review by yasker — APPROVED** [review]

**Pull request merged**